airbyte-cdk/java/airbyte-cdk/db-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/jdbc/SqlOperationsUtils.kt

```diff
@@ -17,136 +16,90 @@ import java.util.*
+                // second loop: bind values to the SQL string.
+                // 1-indexed
+                var i = 1
+                for (message in partition) {
+                    // Airbyte Raw ID
+                    statement.setString(i++, uuidSupplier.get().toString())
 
-        jdbcDatabase.execute(
-            CheckedConsumer { connection: Connection ->
+                    // Message Data
+                    statement.setString(i++, message.serialized)
 
-                // Strategy: We want to use PreparedStatement because it handles binding values to
-                // the SQL query
-                // (e.g. handling formatting timestamps). A PreparedStatement statement is created
-                // by supplying the
-                // full SQL string at creation time. Then subsequently specifying which values are
-                // bound to the
-                // string. Thus there will be two loops below.
-                // 1) Loop over records to build the full string.
-                // 2) Loop over the records and bind the appropriate values to the string.
-                // We also partition the query to run on 10k records at a time, since some DBs set a
-                // max limit on
-                // how many records can be inserted at once
-                // TODO(sherif) this should use a smarter, destination-aware partitioning scheme
-                // instead of 10k by
-                // default
-                for (partition in Iterables.partition(records, 10000)) {
-                    val sql = StringBuilder(insertQueryComponent)
-                    partition.forEach(
-                        Consumer { r: PartialAirbyteMessage? -> sql.append(recordQueryComponent) }
+                    // Extracted At
+                    statement.setTimestamp(
+                        i++,
+                        Timestamp.from(Instant.ofEpochMilli(message.record!!.emittedAt))
                     )
-                    val s = sql.toString()
-                    val s1 = s.substring(0, s.length - 2) + (if (sem) ";" else "")
-
-                    connection.prepareStatement(s1).use { statement ->
-                        // second loop: bind values to the SQL string.
-                        // 1-indexed
-                        var i = 1
-                        for (message in partition) {
-                            // Airbyte Raw ID
-                            statement.setString(i, uuidSupplier.get().toString())
-                            i++
 
-                            // Message Data
-                            statement.setString(i, message.serialized)
-                            i++
-
-                            // Extracted At
-                            statement.setTimestamp(
-                                i,
-                                Timestamp.from(Instant.ofEpochMilli(message.record!!.emittedAt))
+                    if (isDestinationV2) {
```

Comment:
shall we avoid this implicit TypingDedupingFlag anymore in new code, especially in static method references.  probably pass it along with default value as false with multi-arg methods.